src/ucp/proto/proto_common.c

```diff
@@ -661,9 +661,21 @@ void ucp_proto_request_zcopy_completion(uct_completion_t *self)
 
 int ucp_proto_is_short_supported(const ucp_proto_select_param_t *select_param)
 {
-    /* Short protocol requires contig/host */
-    return (select_param->dt_class == UCP_DATATYPE_CONTIG) &&
-           UCP_MEM_IS_HOST(select_param->mem_type);
```

Comment:
how does it work when sending small GPU buffer (for example, 8B)? won't we try to use short fast-path send?

Reply:
In case of small GPU buffer `ucp_proto_short_memtype_params` will set `max_length` to 0. So as I understand we would be able to use short only for empty messages.

Reply:
But then it seem that when memtype cache is not empty (which is always the case now), fast-path short will be completely disabled, also for host memory?

Reply:
As it was discussed, that patch affects only proto initialization flow, so it shouldn't break the logic